build.gradle

```diff
@@ -125,7 +125,7 @@ ext {
         mockito: 'org.mockito:mockito-core:3.1.0',
         mockk: 'io.mockk:mockk:1.10.6',
         mockwebserver: 'com.squareup.okhttp3:mockwebserver:4.9.1',
-        robolectric: 'org.robolectric:robolectric:4.5.1',
+        robolectric: 'org.robolectric:robolectric:4.7.1',
```

Comment:
I had to upgrade this to at least 4.7.1 to get the unit tests running successfully on M1 Mac.

When I tried to run the unit tests, the ones that had a dependency on SQLite would fail with the error:

```
Failed to initialize the local storage adapter for the DataStore plugin.
AmplifyException{message=Failed to initialize the local storage adapter for the DataStore plugin., cause=java.lang.RuntimeException: DataStoreException{message=Error in initializing the SQLiteStorageAdapter, cause=java.lang.RuntimeException: com.almworks.sqlite4java.SQLiteException: [-91] cannot load library: java.lang.UnsatisfiedLinkError: /private/var/folders/76/l6xl3gl11clcj6yd5f0k7ddc0000gr/T/1655928563383-0/libsqlite4java.dylib: dlopen(/private/var/folders/76/l6xl3gl11clcj6yd5f0k7ddc0000gr/T/1655928563383-0/libsqlite4java.dylib, 0x0001): tried: '/private/var/folders/76/l6xl3gl11clcj6yd5f0k7ddc0000gr/T/1655928563383-0/libsqlite4java.dylib' (fat file, but missing compatible architecture (have 'i386,x86_64', need 'arm64e')), recoverySuggestion=See attached exception}, recoverySuggestion=Sorry, we don't have a suggested fix for this error yet.}
```

Looking around, it looks like this was fixed in at least [robolectric 4.7](https://github.com/robolectric/robolectric/releases/tag/robolectric-4.7). Related issue: https://github.com/robolectric/robolectric/issues/6311 . We can also consider going to the current 4.8.1. 4.8 had a lot of changes https://github.com/robolectric/robolectric/releases